include/flatbuffers/flatbuffers.h

```diff
@@ -697,6 +698,17 @@ class vector_downward {
     }
   }
 
+  void reset() {
+    assert(allocator_ != nullptr);
+    if (buf_ != nullptr) {
+      allocator_->deallocate(buf_, reserved_);
+    }
+    reserved_ = (initial_size_ + sizeof(largest_scalar_t) - 1) &
+                ~(sizeof(largest_scalar_t) - 1);
+    buf_ = allocator_->allocate(reserved_);
```

Comment:
I should be more precise: `clear()` when there is a nullptr buffer reallocates, and I think we can remove that and let `make_space` handle the reallocation. If the buffer already exists, we can leave it alone and reset the `cur_` pointer.